src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -100,6 +100,37 @@ public AcquireTokenForClientParameterBuilder WithAzureRegion(bool autoDetectRegi
             return this;
         }
 
+        /// <summary>
+        /// Specifies if the token request should be sent to regional ESTS.
+        /// If set, MSAL tries to auto-detect and use a regional Azure authority. This helps keep the authentication traffic inside the Azure region. 
+        /// If the region cannot be determined (e.g. not running on Azure), MSALClientException is thrown with error code region_discovery_failed.
+        /// This feature requires configuration at tenant level.
+        /// By default the value for this variable is false.
+        /// See https://aka.ms/msal-net-region-discovery for more details.
+        /// </summary>
+        /// <param name="autoDetectRegion"><c>true</c> if the token request should be sent to regional ESTS. The default is <c>false</c>.
+        /// </param>
+        /// <param name="fallbackToGlobal"></param>
+        /// <param name="useRegion"> optional parameter to provide region to MSAL. This parameter will be used along with auto detection of region.
+        /// If the region is auto detected, the provided region will be compared with the detected region and used in telemetry to do analysis on correctness of the region provided.
+        /// If auto region detection fails, the provided region will be used for instance metadata.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public AcquireTokenForClientParameterBuilder WithAzureRegion(bool autoDetectRegion, bool fallbackToGlobal, string useRegion = "")
```

Comment:
I was wondering if we should default the fallbackToGlobal to false. Thoughts @henrik-me?

Reply:
Should fallback be defaulted to true? My understanding is that regional ESTS is a complementary feature to improve request performance. So we'd rather send request to global, than to error and not send it at all?